crates/hyperdrive-math/src/short/close.rs

```diff
@@ -56,28 +56,123 @@ impl State {
-    fn calculate_short_proceeds(
+    /// Calculates the proceeds in shares of closing a short position. This
+    /// takes into account the trading profits, the interest that was
+    /// earned by the short, the flat fee the short pays, and the amount of
+    /// margin that was released by closing the short. The math for the
+    /// short's proceeds in base is given by:
+    ///
+    /// $$
+    /// proceeds = (\frac{c1}{c_0 \cdot c}
+    /// + \text{\phi_f}) \cdot \frac{\Delta y}{c}
+    /// - dz
+    /// $$
+    ///
+    /// We convert the proceeds to shares by dividing by the current vault
+    /// share price. In the event that the interest is negative and
+    /// outweighs the trading profits and margin released, the short's
+    /// proceeds are marked to zero.
+    pub fn calculate_short_proceeds_up(
         &self,
         bond_amount: FixedPoint,
         share_amount: FixedPoint,
         open_vault_share_price: FixedPoint,
         close_vault_share_price: FixedPoint,
-        vault_share_price: FixedPoint,
-        flat_fee: FixedPoint,
     ) -> FixedPoint {
-        let mut bond_factor = bond_amount
-            .mul_div_down(
-                close_vault_share_price,
-                // We round up here do avoid overestimating the share proceeds.
-                open_vault_share_price,
-            )
-            .div_down(vault_share_price);
-        bond_factor += bond_amount.mul_div_down(flat_fee, vault_share_price);
+        // NOTE: Round up to overestimate the short proceeds.
+        //
+        // The total value is the amount of shares that underlies the bonds that
+        // were shorted. The bonds start by being backed 1:1 with base, and the
+        // total value takes into account all of the interest that has accrued
+        // since the short was opened.
+        //
+        // total_value = (c1 / (c0 * c)) * dy
+        let mut total_value = bond_amount
+            .mul_div_up(close_vault_share_price, open_vault_share_price)
+            .div_up(self.vault_share_price());
+
+        // NOTE: Round up to overestimate the short proceeds.
+        //
+        // We increase the total value by the flat fee amount, because it is
+        // included in the total amount of capital underlying the short.
+        total_value += bond_amount.mul_div_up(self.flat_fee(), self.vault_share_price());
+
+        // If the interest is more negative than the trading profits and margin
+        // released, then the short proceeds are marked to zero. Otherwise, we
+        // calculate the proceeds as the sum of the trading proceeds, the
+        // interest proceeds, and the margin released.
+        if total_value > share_amount {
+            // proceeds = (c1 / c0 * c) * dy - dz
+            total_value - share_amount
+        } else {
+            fixed!(0)
+        }
+    }
+
+    /// Returns the derivative of the short proceeds calculation, assuming that the interest is
```

Comment:
I know it's obvious in context, but might be good to say "with respect to the bond amount"